lib/facility.rb

```diff
@@ -1,14 +1,89 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, 
+              :address, 
+              :phone, 
+              :services,
+               :r_vehicle,
+               :total
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_details)
+    @name = facility_details[:name]
+    @address = facility_details[:address]
+    @phone = facility_details[:phone]
     @services = []
+    @r_vehicles = []
+    @total = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
+
+  def registered_vehicles 
+    @r_vehicles
+  end 
+
+  def collected_fees
+    @total
+  end
+
+
+def register_vehicle(vehicle)
+  if @services.include?('Vehicle Registration')
+    vehicle.registration_date = Date.today 
+    @r_vehicles << vehicle
+    if vehicle.vehicle_type == :antique
+      @total += 25 
+      @r_vehicles 
+    elsif vehicle.vehicle_type == :ev 
+      @total += 200
+      @r_vehicles 
+    elsif vehicle.vehicle_type == :regular
+      @total += 100
+      @r_vehicles 
+     # binding.pry
+    end 
+  else  
+      nil 
+   end 
+  end 
+
+
+  def administer_written_test(registrant)
+    if @services.include?('Written Test')
+      if registrant.age >= 16 && registrant.permit == true
+        registrant.license_data[:written] = true
+      else
+        false
+      end
+    else
+      false
+    end
+  end
```

Comment:
(this comment applies to the two mehtods below this as well)

We can combine some logic to slim up this method. We could refactor this to look something like: 

```ruby
  def administer_written_test(registrant)
    if @services.include?('Written Test') && registrant.age >= 16 && registrant.permit == true
        registrant.license_data[:written] = true
     else
        false
    end
end
```